### PR TITLE
3.10 release blog post

### PR DESCRIPTION
**Pull request by christinaforney**
Release dev-rel tracking issue: [#5974](https://github.com/sourcegraph/sourcegraph/issues/5974)

### CODE REVIEW

File: blogposts/announcing-sourcegraph-3.10.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+  <div style="padding:56.25% 0 0 0;position:relative;">
+    <iframe src="https://player.vimeo.com/video/{ID}?color=0CB6F4&amp;title=0&amp;byline=" style="position:absolute;top:0;left:0;width:100%;height:100%;" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe>
+  </div>
+  <p style="text-align: center"><a href="https://vimeo.com/{ID}" target="_blank">View on Vimeo</a></p>
+</p>
+-->
+
+Sourcegraph now supports creating and tracking code-aware search and replace campaigns on GitHub and Bitbucket Server. When running a search and replace campaign, you preview the changes, then generate the corresponding pull requests on your GitHub and/or Bitbucket Server instances. Once created, you can track the PRs to completion in a burndown chart.
+
+Sourcegraph is introducing [Comby syntax](https://comby.dev/#basic-usage) for automation search and replace campaigns. Comby is a code aware search syntax that simplifies finding patterns in code. This makes operations that previously required complex regexp or sed queries much simpler to write. For example, changing how errors are logged througout your code:
```
Comment (christinaforney) [inline blogposts/announcing-sourcegraph-3.10.md]: @rvantonder - can you please review this section?

File: blogposts/announcing-sourcegraph-3.10.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+// match:
+errors.New(fmt.Sprintf(:[args]))
+
+// replace:
+fmt.Errorf(:[args])
+```
+
+Sourcegraph Automation campaigns are in private beta. [Watch the Automation screencasts](https://about.sourcegraph.com/product/automation#see-it-in-action) to see what we have planned and [apply for early access](https://about.sourcegraph.com/contact/request-automation-demo/) to Automation for your organization.
+
+## Precise LSIF-based code intelligence for 5 languages
```
Comment (christinaforney) [inline blogposts/announcing-sourcegraph-3.10.md]: @chrismwendt and @efritz can you please review this section?

File: blogposts/announcing-sourcegraph-3.10.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+  </div>
+  <p style="text-align: center"><a href="https://vimeo.com/{ID}" target="_blank">View on Vimeo</a></p>
+</p>
+-->
+
+With the completion of the Java LSIF indexer, LSIF-based precise code intelligence now supports five languages: Go, Typescript, Java, C++, and Python. See [lsif.dev](https://lsif.dev/) for the most up-to-date information on the list of languages with LSIF indexers.
+
+We are looking for feedback from project owners interested in trying out precise code intelligence! If you are responsible for projects using any of the languages we support, we’d love to hear from you. Use our new [LSIF quickstart guide](https://docs.sourcegraph.com/user/code_intelligence/lsif_quickstart) to try it out manually before adding it as a step in your CI.
+
+## Scaling search for our largest customers
```
Comment (christinaforney) [inline blogposts/announcing-sourcegraph-3.10.md]: @slimsag  can you please review this section (or @keegancsmith when he wakes up :) )

File: blogposts/announcing-sourcegraph-3.10.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+Customers with large Kubernetes deployments can now horizontally scale their search indexes by adding more search nodes.
+
+**This improvement requires a Kubernetes migration**. The Kubernetes [deployment manifest](https://github.com/sourcegraph/deploy-sourcegraph) for indexed-search services has changed from a Normal Service to a Headless Service. This enables Sourcegraph to individually resolve indexed-search pods. Services are immutable, so please follow the [migration guide](https://github.com/sourcegraph/deploy-sourcegraph/blob/master/docs/migrate.md#310).
+
+## Fully automated release testing process
+
+As part of Sourcegraph’s ongoing commitment to providing the highest quality, stable releases, our manual end-to-end release testing process has been replaced with a fully automated regression testing suite. Read more about this in the upcoming blogpost from Beyang, Sourcegraph CTO.
```
Comment (christinaforney) [inline blogposts/announcing-sourcegraph-3.10.md]: @beyang - are you ok with the last sentence commitment? If not we can pull it. Does this section capture what we accomplished? Any implementation details that would be cool to share?

File: blogposts/announcing-sourcegraph-3.10.md [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+</p>
+-->
+
+Autocomplete suggestions make it easier to use filters in your Sourcegraph searches. Use `ctrl-space` to trigger suggestions for filters you can apply to your query. As you type, the suggestions become more relevant, helping you to apply correct filter options or autocomplete repository or file names.
+
+## Sourcegraph provides native code intelligence to GitLab
+
+<p class="container">
+  <div style="padding:56.25% 0 0 0;position:relative;">
+    <iframe src="https://player.vimeo.com/video/372226334/de668e24fa?color=0CB6F4&amp;title=0&amp;byline=" style="position:absolute;top:0;left:0;width:100%;height:100%;" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe>
```
Comment (christinaforney) [inline blogposts/announcing-sourcegraph-3.10.md]: @ryan-blunden - the link to this video isn't working and I'm not sure why.

File: blogposts/announcing-sourcegraph-3.10.md [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+
+Soon, GitLab.com users will be able to opt-in to native code intelligence for public repositories by enabling Sourcegraph in their user preferences. This feature is being rolled out incrementally to public projects on GitLab.com, and will first be available on the [GitLab repository](https://gitlab.com/gitlab-org/gitlab).
+
+Users with private GitLab instances or repositories can enable Sourcegraph code intelligence as of GitLab 12.5. Take a look at the [Sourcegraph integration documentation](https://gitlab.com/gitlab-org/gitlab/blob/master/doc/integration/sourcegraph.md) for how to configure it on your instance.
+
+## Create cross-repository search and replace campaigns
```
Comment (christinaforney) [inline blogposts/announcing-sourcegraph-3.10.md]: @sqs @mrnugget (and anyone else on the a8n team) - are you comfortable with how I framed this section? I wanted to use it as an opportunity to introduce comby syntax to the user, since this is a pretty cool new conceptual model that is very powerful.

File: blogposts/announcing-sourcegraph-3.10.md [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+title: "Sourcegraph 3.10: Improved search autocompletion, native GitLab integration, and find-and-replace automation campaigns"
+author: Christina Forney
+publishDate: 2019-11-20T10:00-07:00
+tags: [blog]
+slug: sourcegraph-3.10
+heroImage: https://about.sourcegraph.com/sourcegraph-mark.png
+published: true
+---
+
+Sourcegraph is the standard developer platform for code search and navigation at many of the largest and most advanced technology companies. With Sourcegraph, every company has access to the same kind of tools that Google and Facebook developers use every day.
```
Comment (slimsag) [inline blogposts/announcing-sourcegraph-3.10.md]: I would invert the order of these two sentences ("Sourcegraph does X, it is popular" instead of "Sourcegraph is popular, it does X")

File: blogposts/announcing-sourcegraph-3.10.md [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+## Sourcegraph provides native code intelligence to GitLab
+
+<p class="container">
+  <div style="padding:56.25% 0 0 0;position:relative;">
+    <iframe src="https://player.vimeo.com/video/372226334/de668e24fa?color=0CB6F4&amp;title=0&amp;byline=" style="position:absolute;top:0;left:0;width:100%;height:100%;" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe>
+  </div>
+  <p style="text-align: center"><a href="https://vimeo.com/372226334/de668e24fa" target="_blank">View on Vimeo</a></p>
+</p>
+
+Sourcegraph users use the [Sourcegraph browser extension](https://docs.sourcegraph.com/integration/browser_extension) to take advantage of go-to-definition and find references on their code hosts. GitLab recognizes the value this brings to developer productivity and is trusting Sourcegraph to provide this functionality natively to their users. Read our [feature announcement](https://about.sourcegraph.com/blog/gitlab-integrates-sourcegraph-code-navigation-and-code-intelligence) and [GitLab’s post](https://about.gitlab.com/blog/2019/11/12/sourcegraph-code-intelligence-integration-for-gitlab/) about collaborating in the open.
```
Comment (slimsag) [inline blogposts/announcing-sourcegraph-3.10.md]:
```suggestion
Sourcegraph users use the [Sourcegraph browser extension](https://docs.sourcegraph.com/integration/browser_extension) to take advantage of go-to-definition and find references on their code hosts. GitLab recognizes the value this brings to developer productivity and has partnered with Sourcegraph to provide this functionality natively to all GitLab users. Read our [feature announcement](https://about.sourcegraph.com/blog/gitlab-integrates-sourcegraph-code-navigation-and-code-intelligence) and [GitLab’s post](https://about.gitlab.com/blog/2019/11/12/sourcegraph-code-intelligence-integration-for-gitlab/) about collaborating in the open.
```

"is trusting Sourcegraph to provide this" introduces the idea we might not be trustworthy to pessimistic readers like myself.

File: blogposts/announcing-sourcegraph-3.10.md (hunk 2 above)
Comment (chrismwendt) [reply blogposts/announcing-sourcegraph-3.10.md]: I've mostly been working on the indexers and docs, so this LGTM from that perspective :+1:

File: blogposts/announcing-sourcegraph-3.10.md (hunk 5 above)
Comment (slimsag) [inline blogposts/announcing-sourcegraph-3.10.md]:
```suggestion
    <iframe src="https://player.vimeo.com/video/372226334?color=0CB6F4&amp;title=0&amp;byline=" style="position:absolute;top:0;left:0;width:100%;height:100%;" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe>
```

This looks like what you want.

File: blogposts/announcing-sourcegraph-3.10.md [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+  <div style="padding:56.25% 0 0 0;position:relative;">
+    <iframe src="https://player.vimeo.com/video/{ID}?color=0CB6F4&amp;title=0&amp;byline=" style="position:absolute;top:0;left:0;width:100%;height:100%;" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe>
+  </div>
+  <p style="text-align: center"><a href="https://vimeo.com/{ID}" target="_blank">View on Vimeo</a></p>
+</p>
+-->
+
+Sourcegraph now supports creating and tracking code-aware search and replace campaigns on GitHub and Bitbucket Server. When running a search and replace campaign, you preview the changes, then generate the corresponding pull requests on your GitHub and/or Bitbucket Server instances. Once created, you can track the PRs to completion in a burndown chart.
+
+Sourcegraph is introducing [Comby syntax](https://comby.dev/#basic-usage) for automation search and replace campaigns. Comby is a code aware search syntax that simplifies finding patterns in code. This makes operations that previously required complex regexp or sed queries much simpler to write. For example, changing how errors are logged througout your code:
```
Comment (slimsag) [inline blogposts/announcing-sourcegraph-3.10.md]:
```suggestion
Sourcegraph is introducing [Comby syntax](https://comby.dev/#basic-usage) for automated search and replace campaigns. Comby is a code aware search syntax that simplifies finding patterns in code. This makes operations that previously required complex regexp or sed queries much simpler to write. For example, changing how errors are logged throughout your code:
```

fixed typos:

s/automation/automated/g
s/througout/throughout/g

File: blogposts/announcing-sourcegraph-3.10.md [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+  <div style="padding:56.25% 0 0 0;position:relative;">
+    <iframe src="https://player.vimeo.com/video/{ID}?color=0CB6F4&amp;title=0&amp;byline=" style="position:absolute;top:0;left:0;width:100%;height:100%;" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe>
+  </div>
+  <p style="text-align: center"><a href="https://vimeo.com/{ID}" target="_blank">View on Vimeo</a></p>
+</p>
+-->
+
+With the completion of the Java LSIF indexer, LSIF-based precise code intelligence now supports five languages: Go, Typescript, Java, C++, and Python. See [lsif.dev](https://lsif.dev/) for the most up-to-date information on the list of languages with LSIF indexers.
+
+We are looking for feedback from project owners interested in trying out precise code intelligence! If you are responsible for projects using any of the languages we support, we’d love to hear from you. Use our new [LSIF quickstart guide](https://docs.sourcegraph.com/user/code_intelligence/lsif_quickstart) to try it out manually before adding it as a step in your CI.
```
Comment (slimsag) [inline blogposts/announcing-sourcegraph-3.10.md]: I would rephrase the responsibility aspect here, it sounds like a large commitment. Maybe say something like "If you would like to try out precise code intelligence for one of the languages we support (requires pushing LSIF data from e.g. your CI pipeline), we'd love to get in touch with you!

File: blogposts/announcing-sourcegraph-3.10.md [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+
+## Scaling search for our largest customers
+
+Customers with large Kubernetes deployments can now horizontally scale their search indexes by adding more search nodes.
```
Comment (slimsag) [inline blogposts/announcing-sourcegraph-3.10.md]:
```suggestion
Customers wanting better indexed search performance or wishing to add many more thousands of repositories to Sourcegraph are now in luck! Sourcegraph's indexed search can now scale horizontally in cluster deployments.
```

File: blogposts/announcing-sourcegraph-3.10.md [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+
+**This improvement requires a Kubernetes migration**. The Kubernetes [deployment manifest](https://github.com/sourcegraph/deploy-sourcegraph) for indexed-search services has changed from a Normal Service to a Headless Service. This enables Sourcegraph to individually resolve indexed-search pods. Services are immutable, so please follow the [migration guide](https://github.com/sourcegraph/deploy-sourcegraph/blob/master/docs/migrate.md#310).
```
Comment (slimsag) [inline blogposts/announcing-sourcegraph-3.10.md]: This needs to be much more front-and-center (maybe out of this section and more near the top or something?). It's not just people wanting this improvement that need a migration -- ALL Kubernetes clusters will need to run this migration, period. I would phrase it something like:

> **IMPORTANT:** Kubernetes clusters will require a manual migration, please see [the migration guide](https://github.com/sourcegraph/deploy-sourcegraph/blob/master/docs/migrate.md#310) for details.

File: blogposts/announcing-sourcegraph-3.10.md [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+
+## Fully automated release testing process
+
+As part of Sourcegraph’s ongoing commitment to providing the highest quality, stable releases, our manual end-to-end release testing process has been replaced with a fully automated regression testing suite. Read more about this in the upcoming blogpost from Beyang, Sourcegraph CTO.
```
Comment (slimsag) [inline blogposts/announcing-sourcegraph-3.10.md]:
```suggestion
As part of Sourcegraph’s ongoing commitment to providing the highest quality, stable releases, our manual QA process has in the last two months been turned into a fully automated regression testing suite. We continue to aggressively invest in automated testing of Sourcegraph to provide extremely stable and reliable releases, and ultimately better features for our users. Look forward to reading more about this in an upcoming blogpost from Sourcegraph CTO, Beyang Liu, [on our blog](https://about.sourcegraph.com/blog/).
```

File: blogposts/announcing-sourcegraph-3.10.md [hunk 14]
```diff
@@ -0,0 +1,198 @@
...
+
+### Fixed
+
+- Support hyphens in Bitbucket Cloud team names. [#6154](https://github.com/sourcegraph/sourcegraph/issues/6154)
+- Server will run `redis-check-aof --fix` on startup to fix corrupted AOF files. [#651](https://github.com/sourcegraph/sourcegraph/issues/651)
+- Authorization provider configuration errors in external services will be shown as site alerts. [#6061](https://github.com/sourcegraph/sourcegraph/issues/6061)
+
+### Removed
+
+## 3.9.4
```
Comment (slimsag) [inline blogposts/announcing-sourcegraph-3.10.md]: Why is 3.9.4 and other older releases mentioned here?

Can we just link to the changelog instead of copying it into our blog posts, I wonder?

File: blogposts/announcing-sourcegraph-3.10.md [hunk 15]
```diff
@@ -0,0 +1,198 @@
...
+
+- [@prayashm](https://github.com/prayashm)
+- [@abeyerpath](https://github.com/abeyerpath)
+- [@gerbal](https://github.com/gerbal)
+- [@jlangston](https://github.com/jlangston)
+- [@425183525](https://github.com/425183525)
+- [@kevachen](https://github.com/kevachen)
+- [@brentshermana](https://github.com/brentshermana)
+
+**Deploy or upgrade:** [Local](https://docs.sourcegraph.com/#quickstart-guide) | [AWS](https://github.com/sourcegraph/deploy-sourcegraph-aws) | [DigitalOcean](https://marketplace.digitalocean.com/apps/sourcegraph?action=deploy&refcode=48dfb3ccb51c) | [Kubernetes cluster](https://github.com/sourcegraph/deploy-sourcegraph)
```
Comment (slimsag) [inline blogposts/announcing-sourcegraph-3.10.md]: We are getting rid of the DigitalOcean one click app for now so we should remove the link to it here.

https://github.com/sourcegraph/sourcegraph/issues/6666

File: blogposts/announcing-sourcegraph-3.10.md [hunk 16]
```diff
@@ -0,0 +1,198 @@
...
+
+From the entire Sourcegraph team ([@srcgraph](https://twitter.com/srcgraph)), happy coding!
```
Comment (slimsag) [inline blogposts/announcing-sourcegraph-3.10.md]:
```suggestion
From the entire [@srcgraph](https://twitter.com/srcgraph) team, happy coding!
```

File: blogposts/announcing-sourcegraph-3.10.md (hunk 3 above)
Comment (slimsag) [reply blogposts/announcing-sourcegraph-3.10.md]: LGTM I left some suggestions inline

File: blogposts/announcing-sourcegraph-3.10.md [hunk 17]
```diff
@@ -0,0 +1,198 @@
...
+<!--
+<p class="container">
+  <div style="padding:56.25% 0 0 0;position:relative;">
+    <iframe src="https://player.vimeo.com/video/{ID}?color=0CB6F4&amp;title=0&amp;byline=" style="position:absolute;top:0;left:0;width:100%;height:100%;" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe>
+  </div>
+  <p style="text-align: center"><a href="https://vimeo.com/{ID}" target="_blank">View on Vimeo</a></p>
+</p>
+-->
+
+Sourcegraph now supports creating and tracking code-aware search and replace campaigns on GitHub and Bitbucket Server. When running a search and replace campaign, you preview the changes, then generate the corresponding pull requests on your GitHub and/or Bitbucket Server instances. Once created, you can track the PRs to completion in a burndown chart.
```
Comment (rvantonder) [inline blogposts/announcing-sourcegraph-3.10.md]:
```suggestion
Sourcegraph now supports creating and tracking code-aware search and replace campaigns on GitHub and Bitbucket Server. When running a search and replace campaign, you preview the changes, then generate the corresponding pull requests on your GitHub or Bitbucket Server instances. Once created, you can track the PRs to completion in a burndown chart.
```

^ nit, and I don't feel strongly about it

File: blogposts/announcing-sourcegraph-3.10.md [hunk 18]
```diff
@@ -0,0 +1,198 @@
...
+
+Sourcegraph is the standard developer platform for code search and navigation at many of the largest and most advanced technology companies. With Sourcegraph, every company has access to the same kind of tools that Google and Facebook developers use every day.
+
+<div style="padding-left: 2rem">
+
+[**🔬 Improved autocompletion for search query filters**](#improved-autocompletion-for-search-query-filters)<br />
+
+[**🦊 Sourcegraph provides native code intelligence to GitLab**](#sourcegraph-provides-native-code-intelligence-to-gitlab)<br />
+
+[**🤖 Create cross-repository search and replace campaigns**](#precise-lsif-based-code-intelligence-for-5-languages)<br />
```
Comment (eseliger) [inline blogposts/announcing-sourcegraph-3.10.md]: this is the same link as below

File: blogposts/announcing-sourcegraph-3.10.md (hunk 9 above)
Comment (rvantonder) [inline blogposts/announcing-sourcegraph-3.10.md]: > Comby is a code aware search syntax

Since comby is a tool and not a syntax, maybe "Comby uses" or "supports" or a code aware search syntax? Alt: `Comby syntax is code aware and ...`

> `code-aware`

I think?

> This makes operations that previously required complex regexp or sed queries much simpler to write.

Not so sure about the "previously", it could create the impression that we somehow had this capability before and now changed it (I follow the intended meaning). It might also be a good idea to communicate that this is new? Maybe: `This new capability makes operations that could require complex or awkward regexp queries much simpler to write.`

^ Not super attached, just giving a high level impression. 

> For example, to change how errors are logged throughout your code:

s/changing/to change

File: blogposts/announcing-sourcegraph-3.10.md [hunk 19]
```diff
@@ -0,0 +1,198 @@
...
+  </div>
+  <p style="text-align: center"><a href="https://vimeo.com/{ID}" target="_blank">View on Vimeo</a></p>
+</p>
+-->
+
+Sourcegraph now supports creating and tracking code-aware search and replace campaigns on GitHub and Bitbucket Server. When running a search and replace campaign, you preview the changes, then generate the corresponding pull requests on your GitHub and/or Bitbucket Server instances. Once created, you can track the PRs to completion in a burndown chart.
+
+Sourcegraph is introducing [Comby syntax](https://comby.dev/#basic-usage) for automation search and replace campaigns. Comby is a code aware search syntax that simplifies finding patterns in code. This makes operations that previously required complex regexp or sed queries much simpler to write. For example, changing how errors are logged througout your code:
+
+```c
```
Comment (rvantonder) [inline blogposts/announcing-sourcegraph-3.10.md]: I like this trick a lot.

File: blogposts/announcing-sourcegraph-3.10.md (hunk 5 above)
Comment (christinaforney) [reply blogposts/announcing-sourcegraph-3.10.md]: Adding in a follow up commit since there are two places for this.

File: blogposts/announcing-sourcegraph-3.10.md (hunk 17 above)
Comment (christinaforney) [reply blogposts/announcing-sourcegraph-3.10.md]: for cleanup I'd prefer the and (rather than the or) because I want to communicate that it can create PRs in BOTH places if they want! :)

File: blogposts/announcing-sourcegraph-3.10.md (hunk 13 above)
Comment (christinaforney) [reply blogposts/announcing-sourcegraph-3.10.md]: This is way better, thanks @slimsag!

File: blogposts/announcing-sourcegraph-3.10.md (hunk 14 above)
Comment (christinaforney) [reply blogposts/announcing-sourcegraph-3.10.md]: In a few releases, there were features included in prior patches that we wanted to make sure were mentioned for anyone who just wanted to skim the whole thing here. I started including the previous patches for that reason, so that this was a representation of "everything that changed since the last release". Thoughts?

File: blogposts/announcing-sourcegraph-3.10.md (hunk 18 above)
Comment (christinaforney) [reply blogposts/announcing-sourcegraph-3.10.md]: Good catch, thanks!

File: blogposts/announcing-sourcegraph-3.10.md (hunk 12 above)
Comment (christinaforney) [reply blogposts/announcing-sourcegraph-3.10.md]: How's this?
image

File: blogposts/announcing-sourcegraph-3.10.md (hunk 9 above)
Comment (christinaforney) [reply blogposts/announcing-sourcegraph-3.10.md]: I referred to it as Comby since that is how the campaign is referenced. Should we be calling this structural search and replace instead? I'm happy to introduce new concepts to users, but want to make sure it's the right one...

File: blogposts/announcing-sourcegraph-3.10.md (hunk 10 above)
Comment (christinaforney) [reply blogposts/announcing-sourcegraph-3.10.md]: I actually don't think that sentence adds value here (now that you point it out). Cutting it - let me know if that helps!

File: blogposts/announcing-sourcegraph-3.10.md (hunk 9 above)
Comment (christinaforney) [reply blogposts/announcing-sourcegraph-3.10.md]: Updated - let me know if the updated resolves everything for you!

File: blogposts/announcing-sourcegraph-3.10.md (hunk 1 above)
Comment (mrnugget) [inline blogposts/announcing-sourcegraph-3.10.md]: You used `code-aware` previously, suggesting this for consistency:

```suggestion
Sourcegraph is introducing [Comby syntax](https://comby.dev/#basic-usage) for automation search and replace campaigns. Comby is a code-aware search syntax that simplifies finding patterns in code. This makes operations that previously required complex regexp or sed queries much simpler to write. For example, changing how errors are logged througout your code:
```

File: blogposts/announcing-sourcegraph-3.10.md [hunk 20]
```diff
@@ -0,0 +1,198 @@
...
+<!--
+<p class="container">
+  <div style="padding:56.25% 0 0 0;position:relative;">
+    <iframe src="https://player.vimeo.com/video/{ID}?color=0CB6F4&amp;title=0&amp;byline=" style="position:absolute;top:0;left:0;width:100%;height:100%;" frameborder="0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe>
+  </div>
+  <p style="text-align: center"><a href="https://vimeo.com/{ID}" target="_blank">View on Vimeo</a></p>
+</p>
+-->
+
+Sourcegraph now supports creating and tracking code-aware search and replace campaigns on GitHub and Bitbucket Server. When running a search and replace campaign, you preview the changes, then generate the corresponding pull requests on your GitHub and/or Bitbucket Server instances. Once created, you can track the PRs to completion in a burndown chart.
```
Comment (mrnugget) [inline blogposts/announcing-sourcegraph-3.10.md]: I think something like this sounds more like "look what you can do!" instead of "here's what you have to do":

```suggestion
Sourcegraph now supports creating and tracking code-aware search and replace campaigns on GitHub and Bitbucket Server. When running a search and replace campaign, you can preview the changes across repositories and then generate the corresponding pull requests on your GitHub and/or Bitbucket Server instances. Once created, you can track the PRs to completion in a burndown chart.
```

File: blogposts/announcing-sourcegraph-3.10.md (hunk 6 above)
Comment (mrnugget) [reply blogposts/announcing-sourcegraph-3.10.md]: Yeah, absolutely comfortable with it :) Also 👍 on giving a pretty cool example.

File: blogposts/announcing-sourcegraph-3.10.md (hunk 14 above)
Comment (christinaforney) [reply blogposts/announcing-sourcegraph-3.10.md]: I'm going to leave this for now, but we should chat about if this should change going forward.

File: blogposts/announcing-sourcegraph-3.10.md (hunk 20 above)
Comment (christinaforney) [reply blogposts/announcing-sourcegraph-3.10.md]: Outdated version, so updated manually!

File: blogposts/announcing-sourcegraph-3.10.md (hunk 1 above)
Comment (christinaforney) [reply blogposts/announcing-sourcegraph-3.10.md]: Outdated version, so updated manually!

File: blogposts/announcing-sourcegraph-3.10.md (hunk 14 above)
Comment (christinaforney) [reply blogposts/announcing-sourcegraph-3.10.md]: I'm going to leave this for now, but we should chat about if this should change going forward.

File: blogposts/announcing-sourcegraph-3.10.md (hunk 7 above)
Comment (christinaforney) [reply blogposts/announcing-sourcegraph-3.10.md]: I'm going to re-write this section for the releases going forward. Thanks for the suggestion!

File: blogposts/announcing-sourcegraph-3.10.md (hunk 4 above)
Comment (christinaforney) [reply blogposts/announcing-sourcegraph-3.10.md]: Confirmed.
